core/provider/service.go

```diff
@@ -602,28 +607,43 @@ func (s *Service) getResources(ctx context.Context, p *domain.Provider) ([]*doma
 			filteredResources = append(filteredResources, r)
 		}
 	}
-
 	flattenedProviderResources := flattenResources(filteredResources)
+
 	existingProviderResources := map[string]bool{}
-	for _, r := range flattenedProviderResources {
-		for _, er := range existingGuardianResources {
-			if er.Type == r.Type && er.URN == r.URN {
-				if existingDetails := er.Details; existingDetails != nil {
-					if r.Details != nil {
+	if resourcesCount != nil {
+		*resourcesCount += len(flattenedProviderResources)
+	}
+	UpdatedResourceCount := 0
+	for _, newResource := range flattenedProviderResources {
+		for _, existingResource := range existingGuardianResources {
+			if existingResource.Type == newResource.Type && existingResource.URN == newResource.URN {
+				if existingDetails := existingResource.Details; existingDetails != nil {
+					if newResource.Details != nil {
 						for key, value := range existingDetails {
-							if _, ok := r.Details[key]; !ok {
-								r.Details[key] = value
+							if _, ok := newResource.Details[key]; !ok {
+								newResource.Details[key] = value
 							}
 						}
 					} else {
-						r.Details = existingDetails
+						newResource.Details = existingDetails
+					}
+					if isUpdated, diff := compareResources(*existingResource, *newResource); isUpdated {
+						s.logger.Info(ctx, "diff", "resources", diff)
+						UpdatedResourceCount++
+						if upsertedResourcesCount != nil {
+							*upsertedResourcesCount++
+						}
+						s.logger.Info(ctx, "Resources is updated", "resource", newResource.Name)
```

Comment:
let's use lowercase for the logs, IIRC in the rest of the code we log in lowercase. Would be better if we can be consistent